File: Ohtap E Commerce Domain Models/Development/NetCore/Models Core/Models Core/Communication/ResponseViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Ohtap.ECommerce.Domain.Models.Core.Communication
{
    public class ResponseViewModel<T, P> : ResponseViewModelBase, IResponseViewModel<T, P>
        where T : IRequestData, new()
        where P : IResponseData, new()
    {
        public T RequestData { set; get; }
        public P ResponseData { set; get; }

        IMessageList<IMessage> IResponseViewModel<T, P>.Messages { get { return this.Messages as IMessageList<IMessage>; } set { this.Messages = (MessageList)value; } }

        public ResponseViewModel() : base()
        {
            this.ResponseData = new P();
        }
        public ResponseViewModel(T RequestData) : base()
        {
            this.RequestData = RequestData;
            this.ResponseData = new P();
        }
    }
}

File: Ohtap E Commerce Domain Models/Development/PHP/Core/Communication/ResponseViewModel.php
<?php
namespace Ohtap\ECommerce\DomainModels\Core\Communication;

class ResponseViewModel implements IResponseViewModel
{
    public  $RequestData;
    public  $ResponseData;
    public  $Messages;
    public  $State;

    public function __construct($RequestData)
    {
        $this->RequestData = $RequestData;
    }
}






